### DISCUSSION TITLE
Kubernetes 1.31 Major Themes

### DISCUSSION THREAD

**Discussion by a-mccarthy** [kubernetes/sig-release#2526] [General]
This is a discussion to collect Major Themes for the v1.31 release.

Major themes are an essential part of the Kubernetes release cycle as they provide a birds-eye view of major improvements that are coming as part of the upcoming release. The release comms team is asking for SIGs to comment on this discussion with items that we should highlight in release communications for 1.31. For each submission, please include a link to the KEP, the stage the feature is moving to, and a brief description of the KEP.

The deadline for 1.31 Major Themes is **Tuesday 23rd July 2024**. 

Here is the [discussion thread](https://github.com/kubernetes/sig-release/discussions/2424) from the last release cycle.

cc @kubernetes/release-team-leads
cc @edithturn @mbianchidev @hailkomputer @rashansmith

**Comment by aojea** [comment]
### SIG Network in 1.31

* [Kube-proxy nftables backend ](https://github.com/kubernetes/enhancements/issues/3866)moves to beta in v1.31. Kube-proxy implements a new backend based on nftables, the replacement of iptables by the Linux Kernel community, aiming to solve the existing performance problems with iptables.
* [Traffic Distribution for Services](https://github.com/kubernetes/enhancements/issues/4444) moves to beta in v1.31. After several iterations on finding the best user experience and traffic engineering capabilities for Services networking, the SIG implemented a new field in the Service specification that serves as preference or hint for the underlying implementation to consider while making routing decisions.
* [Kube-proxy improved ingress connectivity reliability](https://github.com/kubernetes/enhancements/issues/3836) is GA in v1.31. One of the common problems with Load Balancers in Kubernetes is the synchronization between the different components involved to avoid traffic drop, this KEP implements a mechanism in kube-proxt for load balancers to do connection draining for terminating Nodes and establish some best practices for cloud providers and Kubernetes load balancers implementations.
*[ Multiple Service CIDRs](https://github.com/kubernetes/enhancements/issues/1880) moves to beta in v1.31 (disabled by default). This KEP allow users to dynamically resize the Service CIDR range assigned to the cluster, especially useful for cluster that have IP exhaustion problems.

**Reply by a-mccarthy** [reply]
@aojea can you confirm these should all be included as Major themes?

**Reply by aojea** [reply]
updated and confirmed cc: @thockin @danwinship

**Comment by xing-yang** [comment]
SIG Storage enhancements in 1.31:

* [Persistent Volume last phase transition time](https://github.com/kubernetes/enhancements/issues/3762) feature moved to GA in v1.31. This feature adds a PersistentVolumeStatus field which holds a timestamp of when a PersistentVolume last transitioned to a different phase. This allows you to measure time e.g. between a PV Pending and Bound. This can be also useful for providing metrics and SLOs.
* [Always Honor PersistentVolume Reclaim Policy](https://github.com/kubernetes/enhancements/issues/2644) moved to Beta in v1.31. This feature prevents volumes from being leaked by honoring the PV Reclaim policy after the Bound PVC is also deleted.
* [Kubernetes VolumeAttributesClass ModifyVolume](https://github.com/kubernetes/enhancements/issues/3751) moved to Beta in v1.31. This feature extends the Kubernetes Persistent Volume API to allow users to dynamically modify volume options (such as IOPs and throughput), after a volume is provisioned.
cc @saad-ali @msau42 @jsafrane

**Comment by ritazh** [comment]
From SIG-Auth:

- [4193](https://kep.k8s.io/4193) Bound service account token improvements going Beta in 1.31
- [4601](https://kep.k8s.io/4601) Authorize with field and label selectors going Alpha in 1.31
- [4633](https://kep.k8s.io/4633) Only allow anonymous auth for configured endpoints going Alpha in 1.31

cc @liggitt @enj @deads2k @munnerz @vinayakankugoyal

**Comment by SergeyKanzhelev** [comment]
From sig-node:

AI/ML workload support:
1. New DRA APIs for better accelerators and other hardware management: https://github.com/kubernetes/enhancements/issues/3063
2. OCI artifacts as VolumeSource to deliver model weights more efficiently and user friendly: https://github.com/kubernetes/enhancements/issues/4639
3. Better devices errors visibility - the device health is now in pod status: https://github.com/kubernetes/enhancements/issues/4680 

Deprecations and graduations:
1. Cgroup v1 enters the maintenance mode, please switch to use cgroupv2 and report any issues: https://github.com/kubernetes/enhancements/issues/4569
3. AppArmor is now GA. Also since 1.30, AppArmor is controlled via fields, not annotations. Please migrate away from using annotations.

@mrunalp @dchen1107 @derekwaynecarr
